src/types/account-manager.ts

```diff
@@ -0,0 +1,218 @@
+import algosdk from 'algosdk'
+import * as algokit from '..'
+import { AccountInformation, SigningAccount, TransactionSignerAccount } from './account'
+import { AlgoAmount } from './amount'
+import { ClientManager } from './client-manager'
+import { SendTransactionFrom } from './transaction'
+
+/** Creates and keeps track of signing accounts against sending addresses. */
+export class AccountManager {
+  private _clientManager: ClientManager
+  private _accounts: { [address: string]: TransactionSignerAccount } = {}
+  private _defaultSigner?: algosdk.TransactionSigner
+
+  /**
+   * Create a new account creator
+   * @param algod The algod client to use
+   * @param kmd The optional kmd client to use
+   */
+  constructor(clientManager: ClientManager) {
+    this._clientManager = clientManager
+  }
+
+  /**
+   * Sets the default signer to use if no other signer is specified.
+   * @param signer The signer to use, either a `TransactionSigner` or a `TransactionSignerAccount`
+   * @returns The `AccountManager` so method calls can be chained
+   */
+  public withDefaultSigner(signer: algosdk.TransactionSigner | TransactionSignerAccount): AccountManager {
+    this._defaultSigner = 'signer' in signer ? signer.signer : signer
+    return this
+  }
+
+  /**
+   * Records the given account against the address of the account for later retrieval and returns a `TransactionSignerAccount`.
+   * @param account The account to use.
+   * @returns A `TransactionSignerAccount` for the given account.
+   */
+  private signerAccount<T extends SendTransactionFrom>(account: T): TransactionSignerAccount & { account: T } {
```

Comment:
This is a private method that provides a bridge between the underlying AlgoKit account methods that return various subsets of `SendTransactionFrom` and the consistent `TransactionSignerAccount` type this class exposes. It still provides the underlying type via `account` just in case you want to do stuff with that (e.g. pull out `sk` etc.).